src/GafferML/Tensor.cpp

```diff
@@ -213,6 +216,16 @@ Tensor::Tensor( const IECore::ConstDataPtr &data, std::vector<int64_t> shape )
 				std::copy( typedData->readable().begin(), typedData->readable().end(), value.GetTensorMutableData<bool>() );
 				m_state = new State{ std::move( value ), nullptr };
 			}
+			else if constexpr( std::is_same_v<DataType, StringVectorData> )
+			{
+				// Special case for the vector of std::string fiasco.
+				Ort::AllocatorWithDefaultOptions allocator;
+				Ort::Value value = Ort::Value::CreateTensor(
+					allocator, shape.data(), shape.size(), ONNX_TENSOR_ELEMENT_DATA_TYPE_STRING
+				);
+				std::copy( typedData->readable().begin(), typedData->readable().end(), value.GetTensorMutableData<std::string>() );
```

Comment:
I'm really surprised this works. As far as I can see, the underlying ONNX C library represents strings as `const char *`, and there's no guarantee that `std::string` has the same layout (although it is possible to implement it so). Even if it did have the same layout, I think we've leaked the strings here, because the destructors for the copies will never get called (there aren't `std::string` objects in the C library).

I think we really need to build a `vector<const char *`> here and pass it to `value.FillStringTensor()` instead. Does that make sense?

Reply:
Yes that makes sense but this comment is very confusing https://github.com/microsoft/onnxruntime/blob/main/include/onnxruntime/core/session/onnxruntime_c_api.h#L186

Reply:
addressed here c7db62db2b5f16670e52e05837488387164ce711

Reply:
> Yes that makes sense but this comment is very confusing

Oh, that is confusing! Looking at the source it looks like the C library is actually implemented in C++ behind the scenes, so it's plausible that the tensor is backed by `std::string`, which would also explain how this was working before. I'd prefer not to depend on that though and go through the "proper" API as you've done now - thanks for the update.